src/Database/LSMTree/Internal/Chunk.hs

```diff
@@ -0,0 +1,98 @@
+module Database.LSMTree.Internal.Chunk
+(
+    -- * Chunks
+    Chunk (Chunk),
+
+    -- * Balers
+    Baler,
+    createBaler,
+    feedBaler,
+    readBalerRemnant
+)
+where
+
+import           Prelude hiding (concat, length, splitAt)
+
+import           Control.Monad.ST.Strict (ST)
+import           Data.STRef.Strict (STRef, newSTRef, readSTRef, writeSTRef)
+import           Data.Vector.Primitive (Vector, concat, length, slice, splitAt)
+import           Data.Word (Word8)
+
+-- * Chunks
+
+-- | A chunk of bytes, typically output during incremental index serialisation.
+newtype Chunk = Chunk (Vector Word8)
+
+-- * Balers
+
+-- Mutable state of a baler.
+data BalerState = BalerState
+                      ![Vector Word8] -- Reverse list of queued blocks
+                      !Int            -- Total size of queued blocks
+
+{-|
+    An object that receives blocks of bytes and repackages them into chunks of
+    preferably a given maximum size.
+-}
+data Baler s = Baler
+                   !Int                  -- Maximum chunk size
+                   !(STRef s BalerState) -- Reference to the mutable state
+
+-- | Creates a new baler.
+createBaler :: Int            -- ^ Maximum chunk size
```

Comment:
```suggestion
createBaler :: Int            -- ^ Maximum chunk size in bytes
```

Reply:
I deliberately left out this bit, because the documentation of `Chunk` already says that the chunks we’re talking about here are chunks of bytes. However, it might be good to be a bit more explicit. I’ll make this change and also extend the comment in the definition of `Baler` in the same way.

Reply:
Done: 50c287f4531529a921d7e7a22cc0dfd3d228b7da.